[13.0] Improve and refactor shopinvader_import_image

simahawk opened the pull request:
Commits are self-explanatory.

simahawk committed:
import_image: improve zipfile naming

simahawk committed:
import_image: track file not found and skip it

simahawk committed:
import_image: add external storage support

simahawk committed:
import_image: add queue job support

* model refactored as persistent
* added cron to cleanup done imports
* fixed and improved report
* added quick action from backend section

simahawk committed:
import_image: fix product.product import

lmignon reviewed (approved):
LGTM: Code review

i-vyshnevska commented on shopinvader_import_image/models/import_image.py:
```diff
@@ -202,7 +254,8 @@ def _do_import(self, lines, product_model, options=None):
             elif product_model == "product.product":
-                tmpl_id = prod["product_tmpl_id"]
+                # TODO: test product.product import
+                tmpl_id = prod["product_tmpl_id"][0]
```
do we need relly on product_model for values on line 250?

simahawk replied:
yes, because the relation is only on template. 
Actually, the images are always imported at template level and you can manage the relation w/ variant using tags.
BUT we gonna improve this because is not enough. We should be able to compute images on the variant itself automatically.

i-vyshnevska replied:
my point code for domain on line 250 also should be fixed, no?

HviorForgeFlow commented:
I could not test this Pr without changing magic dependency on manifest from magic to python-magic

simahawk committed:
import_image: split jobs by chunk

Before this change all images were imported in the same job.
This can be very heavy beacause all images are fetched and created in the same transaction.

Now you can set the number of items per job and the import will be split across N jobs.

simahawk committed:
import_image: handle product attr value relation

Images can be related to specific variant attribute values
so that the images will be assigned to specific variants
instead to all of them with the same template.

Now the attributes are computed based on the variant
if you import by 'product.product' model.

simahawk committed:
import_image: fix dependency on python-magic

simahawk commented:
> I could not test this Pr without changing magic dependency on manifest from magic to python-magic

yeah we had that in our odoo requirements, I changed it, tnx.

FTR I improved the job handling to work in chunks and added prod attr value handling in last commits :wink:

simahawk commented:
/ocabot merge major
